packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -96,7 +101,7 @@ class ImageStream extends Diagnosticable {
   ImageStreamCompleter get completer => _completer;
   ImageStreamCompleter _completer;
 
-  List<ImageListener> _listeners;
+  Map<ImageListener, ImageErrorListener> _listeners;
```

Comment:
what if the same listener is added twice (maybe with different error listeners)?

Reply:
The latest error listener would precede. It's in the API documentation. I was hoping it's the most predictable behavior.

Reply:
Ah I see what you meant by list of pairs. I think the gist of what you're saying is that no callbacks should ever be removed without calling a `remove*` method. 

Changed to allow duplications in both directions.